### PR TITLE
[Enhancement] Notify Expense Claim Approver whenever Expense Claim is created

### PR DESCRIPTION
**Pull request by KanchanChauhan**

### CODE REVIEW

File: erpnext/hr/doctype/expense_claim/expense_claim.py [hunk 1]
```diff
@@ -36,6 +36,10 @@ def validate(self):
 		if self.task and not self.project:
 			self.project = frappe.db.get_value("Task", self.task, "project")
 
+	def on_update(self):
+		if self.exp_approver and self.approval_status == "Draft":
+			self.notify_expense_claim_approver()
```
Comment (nabinhait) [inline erpnext/hr/doctype/expense_claim/expense_claim.py]: So, this email will go on every save of the Expense Claim, which is not so good. Add a button instead to send the email.

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Comment by nabinhait** [comment]
Reopen after fixing.

**Comment by KanchanChauhan** [comment]
Button added to notify approver.

screen shot 2017-08-03 at 11 34 21 am

**Comment by nabinhait** [comment]
@KanchanChauhan Please add a custom button in the toolbar, it will be shown only in draft mode and if status is Draft.

**Comment by KanchanChauhan** [comment]
screen shot 2017-08-03 at 12 39 44 pm

**Comment by mbauskar** [comment]
@KanchanChauhan,

Thanks for PR.

Travis checks failed please check and fix the test cases. Closing the PR for now but you can reopen it.

**Comment by rmehta** [comment]
Why not Email Alert?

**Comment by nabinhait** [comment]
@rmehta Email Alert will go every time on saving the Expense Claim, which is not desired.

File: erpnext/hr/doctype/expense_claim/expense_claim.js [hunk 2]
```diff
@@ -173,6 +173,17 @@ frappe.ui.form.on("Expense Claim", {
 			}, __("View"));
 		}
 
+		if (frm.doc.docstatus===0 && frm.doc.approval_status=="Draft"
+			&& frm.doc.exp_approver) {
+			frm.add_custom_button(__('Notify Expense Claim Approver'),
```
Comment (nabinhait) [inline erpnext/hr/doctype/expense_claim/expense_claim.js]: Can we change the label to "Notify Approver"?

**Review by nabinhait — CHANGES_REQUESTED** [review]
Also please add a screenshot of the email

**Comment by rmehta** [comment]
@nabinhait I have added a feature to add a flag in Email Alert (like "Email Sent"). If its true, then it will not re-sent 😄

**Comment by nabinhait** [comment]
@rmehta Thats great 👍 But in this case, user can add more expenses after first saving, we really don't know when it is finalised.

**Comment by rmehta** [comment]
Then set a property and trigger the email alert. the message should be configurable...

**Pull request closed without merging**